lib/upipe-modules/upipe_audio_merge.c

```diff
@@ -0,0 +1,618 @@
+}
+
+/** @internal @This processes control commands on an input subpipe of an
+ *  audio_merge pipe.
+ *
+ * @param upipe description structure of the pipe
+ * @param command type of command to process
+ * @param args arguments of the command
+ * @return an error code
+ */
+static int upipe_audio_merge_sub_control(struct upipe *upipe,
+                                         int command, va_list args)
+{
+    UBASE_HANDLED_RETURN(upipe_audio_merge_sub_control_super(upipe, command, args));
+    UBASE_HANDLED_RETURN(upipe_control_provide_request(upipe, command, args));
+    switch (command) {
+        case UPIPE_SET_FLOW_DEF: {
+            struct uref *flow_def = va_arg(args, struct uref *);
+            return upipe_audio_merge_sub_set_flow_def(upipe, flow_def);
+        }
+
+        default:
+            return UBASE_ERR_UNHANDLED;
+    }
+}
+
+/** @internal @Copies data from the input urefs to an output buffer.
+ *
+ * @param upipe description structure of the pipe
+ * @param out_data the output buffer
+ */
+static void upipe_audio_merge_copy_to_output(struct upipe *upipe, float **out_data)
+{
+    int8_t out_channel_count = 0;
+    struct upipe_audio_merge *upipe_audio_merge = upipe_audio_merge_from_upipe(upipe);
+    struct uchain *uchain;
+    int i, j;
+
+    ulist_foreach (&upipe_audio_merge->inputs, uchain) {
+        struct upipe_audio_merge_sub *upipe_audio_merge_sub =
+            upipe_audio_merge_sub_from_uchain(uchain);
+
+        uint8_t planes = 0;
+        UBASE_ERROR(upipe, uref_sound_flow_get_planes(upipe_audio_merge_sub->flow_def, &planes));
+
+        float *in_data[planes];
+        UBASE_ERROR(upipe, uref_sound_read_float(upipe_audio_merge_sub->uref, 0, -1, (const float**)in_data, planes));
+
+        uint64_t input_num_samples = 0;
+        UBASE_ERROR(upipe, uref_sound_flow_get_samples(upipe_audio_merge_sub->uref, &input_num_samples));
+
+        uint64_t output_num_samples = 0;
+        UBASE_ERROR(upipe, uref_sound_flow_get_samples(upipe_audio_merge->flow_def, &output_num_samples));
+
+        uint8_t output_channels = 0;
+        UBASE_ERROR(upipe, uref_sound_flow_get_channels(upipe_audio_merge->flow_def, &output_channels));
+
+        // If the samples of the input uref != our output sample size, throw an error and don't copy
+        if (unlikely(input_num_samples != output_num_samples))
+            upipe_err_va(upipe, "input samples (%"PRIu64") != output samples (%"PRIu64")!",
+                input_num_samples, output_num_samples);
+        // Todo: ensure #planes in uref matches the flowdef?
+        else {
+            for ( i = 0; i < planes; i++ ) {
+                /* Only copy up to the number of channels in the output flowdef,
+                and thus what we've allocated */
+                if ((out_channel_count + i) < output_channels) {
+                    for ( j = 0; j < input_num_samples; j++) {
+                        out_data[out_channel_count+i][j] = in_data[i][j];
+                    }
```

Comment:
remove braces

Reply:
removed all the way up to the `else`, lots of nesting but I think that's the most consistent?